UnityWeld/Binding/Internal/PropertyEndPoint.cs

```diff
@@ -49,6 +49,19 @@ public PropertyEndPoint(object propertyOwner, string propertyName, IAdapter adap
 
             this.propertyName = propertyName;
             this.property = type.GetProperty(propertyName);
+
+            // If the binding property is an AnimatorParameterTrigger,
+            // we change the owner to the instance of the property
+            // and change the property to "TriggerSetOrReset"
+            if(this.property.PropertyType == typeof(AnimatorParameterTrigger))
+            {
+                var animatorParamaterTrigger = this.property.GetValue(propertyOwner, null) as AnimatorParameterTrigger;
+                this.propertyName = "TriggerSetOrReset";
+                this.propertyOwner = animatorParamaterTrigger;
+                type = this.propertyOwner.GetType();
+                this.property = type.GetProperty(this.propertyName);
+            }
+

```

Comment:
I don't think you actually have to modify PropertyEndPoint to get it to work with AnimatorParameterTrigger. propertyName and propertyOwner are both passed into this constructor anyway so we shouldn't be overriding them for specific types of properties. 

Having a single method that does different things in specific circumstances is generally bad as it makes it less obvious what the method does from looking at the interface, and also makes it more error prone due to more code paths. Is there a way we can incorporate this part of the functionality into AnimatorParameterTrigger instead?